tests/wicked/2nics/sut/t03_setup_second_card.pm

```diff
@@ -12,29 +12,32 @@
 #             Jose Lausuch <jalausuch@suse.com>
 #             Clemens Famulla-Conrad <cfamullaconrad@suse.de>
 
-use base 'wickedbase';
-use strict;
-use warnings;
+use Mojo::Base 'wickedbase';
 use testapi;
-use utils 'file_content_replace';
+use lockapi;
+use utils qw(file_content_replace systemctl);
 use network_utils 'iface';
 
 
 sub run {
     my ($self, $ctx) = @_;
-    my $config  = '/etc/sysconfig/network/ifcfg-' . $ctx->iface();
-    my $config2 = '/etc/sysconfig/network/ifcfg-' . $ctx->iface2();
+    mutex_wait('t03_dhcpd_setup_complete');
+    my $cfg_ifc1 = '/etc/sysconfig/network/ifcfg-' . $ctx->iface();
+    my $cfg_ifc2 = '/etc/sysconfig/network/ifcfg-' . $ctx->iface2();
     record_info('Info', 'Set up a second card');
-    $self->get_from_data('wicked/static_address/ifcfg-eth0_second_card', $config);
-    $self->get_from_data('wicked/dynamic_address/ifcfg-eth0',            $config2);
+    $self->get_from_data('wicked/dynamic_address/ifcfg-eth0', $cfg_ifc1);
+    $self->get_from_data('wicked/static_address/ifcfg-eth0',  $cfg_ifc2);
+    $self->wicked_command('ifdown', 'all');
+    systemctl('restart wicked');
```

Comment:
I added this `restart wicked` as I had problems to get the IP from DHCP server on eth0.
But I just run a test without it and it worked: http://cfconrad-vm.qa.suse.de/tests/4035

Reply:
if we need wicked restart to get IP we need confirmation from Marius that this would still will be valid test case . looks like workaround inside a test

Reply:
I removed it...